### PR TITLE
fix(richgeo): DD4hep CellID type

### PR DESCRIPTION
**Pull request by c-dilks**
### Briefly, what does this PR introduce?
fix https://github.com/eic/EICrecon/issues/738

### What kind of change does this PR introduce?
- [x] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
no

### Does this PR change default behavior?
no

### CODE REVIEW

**Comment by c-dilks** [comment]
Confirmed EICrecon builds with no errors with DD4hep `master`; did not check if it runs.

File: src/services/geometry/richgeo/RichGeo.h [hunk 1]
```diff
@@ -7,9 +7,12 @@
 #include <iostream>
 #include <fmt/format.h>
 #include <spdlog/spdlog.h>
+#include <edm4hep/SimTrackerHitData.h>
 
 namespace richgeo {
 
+  typedef decltype(edm4hep::SimTrackerHitData::cellID) CellIDType;
```
Comment (wdconinc) [inline src/services/geometry/richgeo/RichGeo.h:14]: `using` is preferred in modern C++.
```suggestion
  using CellIDType = decltype(edm4hep::SimTrackerHitData::cellID);
```
Ref. https://www.nextptr.com/tutorial/ta1193988140/how-cplusplus-using-or-aliasdeclaration-is-better-than-typedef (just a first hit on google)

Comment (wdconinc) [reply src/services/geometry/richgeo/RichGeo.h:14]: I also think `*_t` as a type alias is a more typical naming convention.

Comment (wdconinc) [reply src/services/geometry/richgeo/RichGeo.h:14]: So maybe `cellID_t` or something?

Comment (c-dilks) [reply src/services/geometry/richgeo/RichGeo.h]: It doesn't seem like we have any convention for alias names here in EICrecon.

Comment (wdconinc) [reply src/services/geometry/richgeo/RichGeo.h]: Yeah, I don't feel strongly about this. Absent formatting guidance, I'll resolve this.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**